### ISSUE TITLE
Seeking downloads all the fragments from the start

### ISSUE THREAD

**Issue by chrisrng** [videojs/http-streaming#939]
## Description
When seeking we start downloading fragments from the beginning of the video before playing. It also seems like audio is not working. Does not happen in Safari.

Tested in both
- https://videojs.github.io/http-streaming/
- https://videojs-http-streaming.netlify.app/?debug=false&autoplay=false&muted=false&minified=false&liveui=false&partial=false&url=https%3A%2F%2Flivectorprodmedia12-usea.licdn.com%2F77c9399a-f3fb-4b0d-84fb-7a3c6c5eec9f%2FL4D5cfffe259c40d000-livemanifest.ism%2Fmanifest(format%3Dm3u8-aapl-v3)&type=application%2Fx-mpegurl&keysystems=

## Sources
`https://livectorprodmedia12-usea.licdn.com/77c9399a-f3fb-4b0d-84fb-7a3c6c5eec9f/L4D5cfffe259c40d000-livemanifest.ism/manifest(format=m3u8-aapl-v3)`

## Steps to reproduce
Explain in detail the exact steps necessary to reproduce the issue.

1. Load the video in demo page
2. Seek to the middle
3. Observe in console we download all the fragments of the video before playing

### Expected
We are able to seek and play the video from that point in time

### Error output
No errors on console

## Additional Information
A couple lines from the manifest (validated by apple's mediastreamvalidator)

```
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-PLAYLIST-TYPE:VOD
#EXT-X-ALLOW-CACHE:NO
#EXT-X-MEDIA-SEQUENCE:53
#EXT-X-TARGETDURATION:4
#EXT-X-PROGRAM-DATE-TIME:2020-08-18T05:14:58.106Z
#EXTINF:2.002000,no-desc
Fragments(video=9596970,format=m3u8-aapl-v3,audiotrack=audio_und)
#EXTINF:2.002000,no-desc
Fragments(video=9777150,format=m3u8-aapl-v3,audiotrack=audio_und)
#EXTINF:2.002000,no-desc
Fragments(video=9957330,format=m3u8-aapl-v3,audiotrack=audio_und)
```

### videojs-http-streaming version
what version of videojs-http-streaming does this occur with?
videojs-http-streaming 1.13.3

### videojs version
what version of videojs does this occur with?
video.js 7.9.3

### Browsers
what browsers are affected? please include browser and version for each
* Chrome
* Firefox

Not Safari

### Platforms
what platforms are affected? please include operating system and version or device and version for each
* Mac OSX

**Comment by gkatsev** [comment]
Safari plays back natively and their HLS support is the gold standard.

I do not see exessive downloads when loading via the netlify page. Looking at the manifest I can see that the target duration is 4s but each segment is around 2s. I do know that we have seeking issues in streams like these where you seek over multiple hours and there is a large difference between the segment duration and target duration.

**Comment by chrisrng** [comment]
Hmm you're right I don't see excessive downloads in netlify – but it freezes for me on like the 3rd second of playing (no audio either)

Lemme see if we can reduce the difference between the segment duration and target duration (i know we had some encoder issues)

**Comment by evanfarina** [comment]
@chrisrng @gkatsev I have a couple of updates here, but will need a hand with the buffer issue (see below). The issue as to why segments are being downloaded from the beginning of the video on seek is simple: There is a [bug in segment-loader.js](https://github.com/videojs/http-streaming/blob/v1.13.4/src/segment-loader.js#L855) which incorrectly uses the lastBufferedEnd time as the start time in all cases (good news, this is fixed in the 2.0 release). In the case of seek, the lastBufferedTime refers to the time before the seek. Then, we have the concept of the buffer. On seek, the buffer is flushed. Videojs will try to load segments until the buffer is filled (around 30 seconds of buffer time by default). This further exacerbates the issue because not only are we loading segments from the beginning of the video, we need to load an additional 30 seconds of video once we've caught up.

Lastly, there seems to be a bug in Transmuxer that I'm unable to get to the bottom of. I've observed that, post seek, calls to [this.sourceUpdater_.appendBuffer](https://github.com/videojs/http-streaming/blob/v1.13.4/src/segment-loader.js#L1350) have no effect -- the buffer remains empty. This causes the player to continue to fetch segments until the stream has ended, which can be an eternity if the stream is especially long. @gkatsev - Debugging the Transmuxer is especially complicated and I have no context around that code. Is anyone from the team available to help take a look?

**Comment by gkatsev** [comment]
@evanfarina hey, is that appendBuffer issue you mentioned in your last comment still an issue? If so, can you open a new issue for that?

Going to close this ticket as the original issue has been addressed.
